src/uk/gov/defra/ffc/DefraUtils.groovy

```diff
@@ -7,6 +7,126 @@ def repoUrl = ''
+            // terragrunt destroy
+            echo "HORROR!!! destroy -var \"pr_code=${parameters["pr_code"]}\" -auto-approve"
+            sh "cd london/eu-west-2/ffc/pr${parameters["pr_code"]} ; terragrunt destroy -var \"pr_code=${parameters["pr_code"]}\" -auto-approve"
+            // delete the pr dir
+            echo "cd london/eu-west-2/ffc/ ; git rm -fr pr${parameters["pr_code"]}"
+            sh "cd london/eu-west-2/ffc/ ; git rm -fr pr${parameters["pr_code"]}"
+            // commit the changes back
+            echo "git commit -am \"Delete PR${parameters["pr_code"]} SQS queue config\" ; git push --set-upstream origin master"
+            sh "git commit -am \"Delete PR${parameters["pr_code"]} SQS queue config\" ; git push --set-upstream origin master"
+            echo "infrastructure successfully destroyed"
+            // Recursively delete the current dir (which should be terragrunt in the current job workspace)
+            deleteDir()
+          }
+          break;
+        default:
+          error("destroyInfrastructure error: unsupported item ${item}")
+      }
+    } else {
+      error("destroyInfrastructure error: unsupported target ${target}")
+    }
+  }
+}
+
+def provisionInfrastructure(target, item, parameters) {
+  sshagent(['helm-chart-creds']) {
+    echo "provisionInfrastructure"
+    if (target.toLowerCase() == "aws") {
+      switch (item) {
+        case "sqs":
+          assert __hasKeys(parameters, [['service': ['code', 'name', 'type']], 'pr_code', 'queue_purpose', 'repo_name']) :
+            "parameters should specify pr_code, queue_purpose, repo_name as well as service details (code, name and type)";
+          assert parameters['repo_name'].size() + parameters['pr_code'].toString().size() + parameters['queue_purpose'].size() < 76 :
+            "repo name, pr code and queue purpose parameters should have fewer than 76 characters when combined";
+          dir('terragrunt') {
+            sh "pwd"
+            echo "cloning terraform repo"
+            // git clone repo...
+            git credentialsId: 'helm-chart-creds', url: 'git@gitlab.ffc.aws-int.defra.cloud:terraform_sqs_pipelines/terragrunt_sqs_queues.git'
+
+            dir('london/eu-west-2/ffc') {
+              def dirName = "${parameters["repo_name"]}-pr${parameters["pr_code"]}"
+              if (!fileExists("${dirName}/terraform.tfvars")) {
+                echo "${dirName} directory doesn't exist, creating..."
+                echo "copy queue dir into new dir"
+                // cd into repo, copy queue dir into new dir...
+                sh "cp -fr standard_sqs_queues ${dirName}"
+                echo "adding new dir to git repo"
+                sh "git add ${dirName} ; git commit -m \"${dirName}\" ; git push --set-upstream origin master"
+              }
+              def varFileName = "vars-${(new Date()).getTime().toString()}.tfvars"
+              dir(dirName) {
```

Comment:
This will create a file `vars.tfvars` file every time `provisionInfrastructure` is run (which may be every time the pipeline runs) even if no changes have been made. Is this intended behaviour?